3D UNet for Pelvis Dataset

borova-siska opened the pull request:
Included in the folder is my implementation of a 3D-unet for the pelvis MRI scans dataset. It includes driver, model and restore_model files that can be run for their respective functions. There is also a README.md file for other information.

Commit:
created barebones train.py

Created a script that will load the first image and its labels and segment other parts of itself by using small patches of the image.

Commit:
added plotfigs

Fixed metrics, added some plt code to generate some loss and dice graphs.

Commit:
added example test slice plot

Commit:
fixed a validation issue

Commit:
testing training on one image and validating on another

Commented code to be uncommented later.

Commit:
expanded data to full dataset

Removed patchify functions, wrote a Dataset map function to get the image/label from a path.

Commit:
added test/train/val split

To avoid data leakage, had to write a regex function to group the cases together and split the dataset based on that.

Commit:
cleaned code

Moved some variables around and added comments to make the code a bit more readable.

Commit:
added restore_model script

Adding goliath script for restoring a saved model and saving the labelling on the first test image.

Commit:
added own unet3d model

Changed block comments to inline, looks cleaner? Added some formatting. Removed segmentaiton_models_3D and created my own. Fixed dice coefficient.

Commit:
split model and train

Commit:
made code more functional and added docs

Also added images of model performance.

Commit:
renamed train.py, updated restore_model.py

Commit:
created README.md

Commit:
created empty folder for dataset

Commit:
fixed restore_model.py

Commit:
Merge branch 'topic-recognition' of https://github.com/borova-siska/PatternFlow into topic-recognition

wooboyeong commented:
Code
* dice_coefficient is not calculating Dice for each class
* Why 3 channels for input image? - only need 1 channel
* Did you normalise the images?

README
* Please report Dice score for each class
* When showing example outputs, use a slice where all 6 labels are visible

General Comments
* Commit messages OK
* Pull request OK

Commit:
updated code in response to the suggestions

Updated dice coefficient to account for each class. Reduced 3 channels to 1 channel (should increase efficiency). Added image normalisation to pre-processing.

shakes76 commented:
## TF/Torch Usage
Adequate use and implementation
## Good Practice (Design/Commenting)
Good spacing and comments
Header blocks
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present 
No Data leakage 
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No feedback required
Request Description could be more informative -1
